### ISSUE TITLE
Add Gemma 3

### ISSUE THREAD

**Issue by jackzhxng** [pytorch/executorch#9250]
### 🚀 The feature, motivation and pitch

Add support for the text decoder backbone of the new [Gemma 3](https://github.com/google-deepmind/gemma) (1B / 4B for edge) model.

The model architecture should slide right into our `llama_transformer.py`, with the exception of the interspersed sliding window local attention layers as specified in the [technical report](https://storage.googleapis.com/deepmind-media/gemma/Gemma3Report.pdf), which will require some modifications to our model code. Luckily the way this is done is through [slicing the attention mask](https://github.com/huggingface/transformers/blob/1c287aecfcf94f569eb8740be5be43bda6750682/src/transformers/models/gemma3/modeling_gemma3.py#L411-L431) and not the kv cache, so we can keep using our static kv cache implementation.

Checkpoints are on HuggingFace:
- [google/gemma-3-4b-it](https://huggingface.co/google/gemma-3-4b-it)
- [google/gemma-3-1b-it](https://huggingface.co/google/gemma-3-1b-it)
